parlai/tasks/wrapper/agents.py

```diff
@@ -144,3 +129,33 @@ def shutdown(self):
 
     def update_counters(self):
         self.task.update_counters()
+
+
+class LabelToTextTeacher(AbstractWrapperTeacher):
+    """
+    Teacher that will shift message['labels'][0] into message['text'] for whatever task
+    is specified with --wrapper-task.
+
+    Because the dialogue history is effectively overwritten by this action, all episodes
+    will be flattened into one example each.
+    """
+
+    def __init__(self, opt: Opt, shared=None):
+        super().__init__(opt, shared)
+
+    def act(self):
+        """
+        Act on the previous observation.
+        """
+        act = self.task.act()
+        new_act = copy.deepcopy(act)
+        if 'labels' in act or 'eval_labels' in act:
```

Comment:
I like Emily's suggestion - this is cleaner :) In fact, you could clean it further by doing
```
labels_type = 'labels' if 'labels' in act else 'eval_labels'
labels = act[labels_type]            
```

Reply:
done :)